### PR TITLE
[jenkins] Update defaultenv ios/tvos/osx-x86_64

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
This brings all 4 apple targets to use the same Xcode version by default.

## Motivation and context
More consistent management of jenkins builds

Expectation is clang warning limits will be hit for osx due to new deprecation warnings over 11.3.1. Initially ill just bump the warning threshold, but we'll look to reduce the deprecation usage as appropriate.

## How has this been tested?
Jenkins ;)

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by kambala-decapitator — APPROVED** [review]

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by kambala-decapitator** [comment]
Jenkins build this please

File: docs/README.iOS.md [hunk 1]
```diff
@@ -58,9 +58,10 @@ Several different strategies are used to draw your attention to certain pieces o
 * Device with **iOS 11.0 or newer** to install Kodi after build.
 
 Building for iOS should work with the following constellations of Xcode and macOS versions:
-  * Xcode 9.x against iOS SDK 11.x on 10.13.x (High Sierra)(recommended)
-  * Xcode 9.x against iOS SDK 11.x on 10.14.x (Mojave)(recommended)
+  * Xcode 12.4 against iOS SDK 14.4 on 10.15.7 (Catalina)(recommended)(CI)
+  * Xcode 13.x against iOS SDK 15.5 on 12.x (Monterey)(recommended
```
Comment (kambala-decapitator) [inline docs/README.iOS.md:62]: please close the bracket :)

**Review by kambala-decapitator — APPROVED** [review]
in 37f8f0bc9f1c55be64c65c2b919667f17d43832f commit message you might want to fix typo: Xcode 14+**x**

**Pull request merged**